apps/prairielearn/assets/scripts/instructorAssessmentQuestionsClient.ts

```diff
@@ -12,13 +82,562 @@ onDocumentReady(() => {
+      ? (e.target as HTMLElement).closest('button')?.dataset.alternativeNumber === 'group'
+        ? 'group'
+        : parseInt((e.target as HTMLElement).closest('button')?.dataset.alternativeNumber ?? '0')
+      : null;
+    if (
+      zoneNumber === zones.length - 1 &&
+      questionNumber === zones[zoneNumber].questions.length - 1 &&
+      alternativeNumber === null
+    ) {
+      return;
+    }
+    if (typeof alternativeNumber === 'number') {
+      if (
+        alternativeNumber ===
+        zones[zoneNumber].questions[questionNumber].alternatives.length - 1
+      ) {
+        zones[zoneNumber].questions[questionNumber].alternatives[
+          alternativeNumber
+        ].is_alternative_group = false;
+        zones[zoneNumber].questions.splice(
+          questionNumber + 1,
+          0,
+          zones[zoneNumber].questions[questionNumber].alternatives.pop(),
+        );
+        renumberQuestions();
+        refreshTable();
+        return;
+      } else {
+        const question =
+          zones[zoneNumber].questions[questionNumber].alternatives[alternativeNumber];
+        zones[zoneNumber].questions[questionNumber].alternatives[alternativeNumber] =
+          zones[zoneNumber].questions[questionNumber].alternatives[alternativeNumber + 1];
+        zones[zoneNumber].questions[questionNumber].alternatives[alternativeNumber + 1] = question;
+        renumberQuestions();
+        refreshTable();
+        return;
+      }
+    }
+    if (questionNumber === zones[zoneNumber].questions.length - 1) {
+      zones[zoneNumber + 1].questions.unshift(zones[zoneNumber].questions.pop());
+      refreshTable();
+      return;
+    }
+
+    if (
+      zones[zoneNumber].questions[questionNumber + 1].is_alternative_group &&
+      alternativeNumber !== 'group'
+    ) {
+      zones[zoneNumber].questions[questionNumber].is_alternative_group = true;
+      zones[zoneNumber].questions[questionNumber + 1].alternatives.unshift(
+        zones[zoneNumber].questions[questionNumber],
+      );
+      zones[zoneNumber].questions.splice(questionNumber, 1);
+      renumberQuestions();
+      refreshTable();
+      return;
+    }
+
+    swapQuestions(zoneNumber, questionNumber, questionNumber + 1);
+    return;
+  });
+
+  on('click', '.deleteQuestionButton', (e) => {
+    const deleteButton = (e.target as HTMLElement).closest('button');
+    const zoneNumber = parseInt(deleteButton?.dataset.zoneNumber ?? '0');
+    const questionNumber = parseInt(deleteButton?.dataset.questionNumber ?? '0');
+    const alternativeNumber = parseInt(deleteButton?.dataset.alternativeNumber ?? '0');
+    $('#deleteQuestionModal').replaceWith(
+      (document.createElement('div').innerHTML = DeleteQuestionModal({
+        zoneNumber,
+        questionNumber,
+        alternativeNumber,
+      }).toString()),
+    );
+    $('#deleteQuestionModal').modal('show');
+  });
+
+  on('click', '#confirmDeleteButton', () => {
```

Comment:
I'd encourage you to use `js-*` classes instead of `camelCaseClasses` or IDs.

Reply:
Now using `js-*` classes in [3c6791f](https://github.com/PrairieLearn/PrairieLearn/pull/10255/commits/3c6791f5b7ef00282d313997acf9af8418da3660).